requirements-doc.txt

```diff
@@ -0,0 +1,32 @@
+# Currently we force rdt to use numpy<2.0.0 to build the documentation
+# since the dtcwt and spgl1 are not yet compatible with numpy=2.0.0
+numpy>=1.21.0,<2.0.0
+scipy>=1.11.0
+--extra-index-url https://download.pytorch.org/whl/cpu
```

Comment:
Just a heads up as this is something that has been annoying me as of late: `--extra-index-url` is a [**global** option](https://pip.pypa.io/en/stable/reference/requirements-file-format/#global-options) (doesn't matter where in the requirements.txt file is is specified) and has [**no predefined priority over the index URL**](https://pip.pypa.io/en/stable/cli/pip_install/#finding-packages) (either standard index URL or an alternative one as specified by `-i` or `--index-url`).

Therefore, as per the specs, it is entirely possible that a `torch` from PyPI will be selected instead of the GPU version... or that a numpy from Pytorch (if it existed) would be selected instead of the PyPI one. As far as I have tested this in the past, this situation does not arise.

There is [ongoing discussion](https://github.com/pypa/pip/issues/8606) about setting channel priorities through `pip`, and `uv` [uses a different logic to disambiguate](https://github.com/astral-sh/uv/blob/main/PIP_COMPATIBILITY.md#packages-that-exist-on-multiple-indexes). As far as workarounds go, what I am currently doing in other projects is to create a separate `requirements-torch.txt` with a single `--index-url` (overriding PyPI) and only adding torch-related packaged to that file. Something to consider!

Reply:
Oh interesting, I was not aware of this... let me open an Issue and try to implement the logic you suggest in a new PR